### PR TITLE
Update approach for fetching pairs in Agent 9

### PR DESCRIPTION
**Pull request by e1Ru1o**
# Update approach for fetching pairs  in Agent 9

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] New Agent
- [ ] Fix an agent error
- [x] Update an agent logic/test/documentation
- [ ] Code style update (formatting, renaming)
- [ ] Other (please describe): 

## Comments
Updating approach for validating the impossible pairs.
- Previous approach: All addresses loaded into memory
- Current approach: Fetched `token0` & `token1` from pairs to validate if the pair is registered in the Factory.

### CODE REVIEW

**Review by Kalzak — CHANGES_REQUESTED** [review]

File: Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/abi.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
 import { Interface } from "@ethersproject/abi";
 
 const FACTORY: Interface = new Interface([
-  "function allPairsLength() view returns (uint256)",
-  "function allPairs(uint256 index) view returns (address)",
   "event PairCreated(address indexed token0, address indexed token1, address pair, uint256)",
```
Comment (yasmine9) [inline Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/abi.ts:4]: I think the `PairCreated` event no longer used

File: Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/pairs.fetcher.spec.ts [hunk 2]
```diff
@@ -4,86 +4,109 @@ import { createAddress } from "forta-agent-tools";
...
-      expect(fetcher.factory).toStrictEqual(addr);
+      expect(fetcher.factoryAddress).toStrictEqual(addr);
+    }
+  });
+
+  it("should return false in calls to isImpossiblePair for address that doesn't have token0", async () => {
+    for(let [addr, block,,,token1] of CASES){
+      mockProvider
+        .addCallTo(addr, block, abi.PAIR, "token1", { inputs:[], outputs:[token1]});
+      // no mock initialization for token0 so calls to token1 should return undefined
```
Comment (yasmine9) [inline Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/pairs.fetcher.spec.ts:37]:
```suggestion
      // no mock initialization for token0 so calls to token0 should return undefined
```

File: Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/pairs.fetcher.spec.ts [hunk 3]
```diff
@@ -4,86 +4,109 @@ import { createAddress } from "forta-agent-tools";
...
-    mockProvider.addCallTo(PAIRS[0], 30, abi.PAIR, "getReserves", {
+    mockProvider.addCallTo(CASES[0][0], 30, abi.PAIR, "getReserves", {
       inputs: [],
       outputs: [5, 10, 0],
     });
-    mockProvider.addCallTo(PAIRS[2], 53, abi.PAIR, "getReserves", {
+    mockProvider.addCallTo(CASES[2][0], 53, abi.PAIR, "getReserves", {
       inputs: [],
       outputs: [1, 3, 0],
     });
```
Comment (yasmine9) [inline Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/pairs.fetcher.spec.ts:91]: Why there's a 3rd value in the output? `getReserves` function returns 2 values

**Review by yasmine9 — CHANGES_REQUESTED** [review]

File: Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/pairs.fetcher.spec.ts (hunk 3 above)
Comment (e1Ru1o) [reply Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/pairs.fetcher.spec.ts:91]: Yes `PairFetcher.getReserves` returns 2 values, but the contract signature returns a 3rd one that is the timestamp of the latest update. 
Check here the signature:
https://bscscan.com/address/0x438A242B9774c6D562d43fA2c1410dc9d9dE3281#code#L70

Comment (yasmine9) [reply Impossible-Agent-Suite/LargeDepositWithdrawOnPairs/src/pairs.fetcher.spec.ts:91]: I got it, thanks for the clarification

**Review by yasmine9 — APPROVED** [review]

**Pull request merged**